### PR TITLE
fix micro-hangs

### PR DESCRIPTION
**Pull request by r10s**
testing out if that removes the one-second-hang i am getting from time to time on my larger account ...

### CODE REVIEW

**Comment by r10s** [comment]
while first commit, "async calls to markseen/marknoticed, update global-badge by event", seems to be useful, it does not solve the problem.

meanwhile i find a way to reprocude it:
- Alice sends a contact request to Bob
- Bob accepts the request -> chat opens - and is unresponsive for a second.

the send message may be a single word only, and it also happens when read-receipts are disabled. needs further investigation.

**Comment by r10s** [comment]
as discussed in a/v, i was using the wrong function to start a background thread :) 

the initially used `DispatchQueue.main.async` just delayes execution in the main/ui thread,  `DispatchQueue.global(qos: .background).async` really does the things in another thread.

at a first glance, this seems to work now, then Alice/Bob scenario described above is no longer an issue with the second commit  :)

File: DcCore/DcCore/DC/events.swift [hunk 1]
```diff
@@ -105,6 +105,9 @@ public func handleEvent(event: DcEvent) {
                     "chat_id": Int(data1),
                 ]
             )
+
+            let array = DcContext.shared.getFreshMessages()
+            UIApplication.shared.applicationIconBadgeNumber = array.count
```
Comment (cyBerta) [inline DcCore/DcCore/DC/events.swift:110]: nice catch

File: deltachat-ios/Chat/ChatViewController.swift [hunk 2]
```diff
@@ -265,11 +265,13 @@ class ChatViewController: UITableViewController {
...
         AppStateRestorer.shared.storeLastActiveChat(chatId: chatId)
+
         // things that do not affect the chatview
         // and are delayed after the view is displayed
-        dcContext.marknoticedChat(chatId: chatId)
-        let array = dcContext.getFreshMessages()
-        UIApplication.shared.applicationIconBadgeNumber = array.count
+        DispatchQueue.global(qos: .background).async { [weak self] in
+            guard let self = self else { return }
+            self.dcContext.marknoticedChat(chatId: self.chatId)
```
Comment (cyBerta) [inline deltachat-ios/Chat/ChatViewController.swift:273]: you can use self? here, too

Comment (r10s) [reply deltachat-ios/Chat/ChatViewController.swift:273]: that would result in `self?.dcContext.marknoticedChat(chatId: self?.chatId ?? 0)` - i do not think, this is a big win in readablility.

Comment (cyBerta) [reply deltachat-ios/Chat/ChatViewController.swift:273]: You're right I oversaw the second self there

**Review by cyBerta — APPROVED** [review]
lgtm

**Pull request merged**